src/character.h

```diff
@@ -619,6 +620,31 @@ class Character : public Creature, public visitable<Character>
         // gets add and mult value from enchantment cache
         double calculate_by_enchantment( double modify, enchantment::mod value,
                                          bool round_output = false ) const;
+
+        /** Returns true if the player has any martial arts buffs attached */
```

Comment:
This is an improvement right now, but ideally things outside of player wouldn't introspect into where bonuses come from, they would just get the needed character stats, and character would tabulate stuff based on matrial arts as necessary.